modules/core/src/tools/file/vpIoTools.cpp

```diff
@@ -85,7 +85,7 @@
 
 // Detect endianness of the host machine
 // Reference: http://www.boost.org/doc/libs/1_36_0/boost/detail/endian.hpp
-#if defined(__GLIBC__)
+#if defined(__GLIBC__) || (defined(__GNUC__) && defined(__BYTE_ORDER__))
```

Comment:
Was not able to detect the endianness on Alpine and Raspi4.
Taken from:

https://github.com/rofl0r/musl/blob/d05aaedaabd4f5472c233dbbd1ff4bb9c9c99794/include/endian.h#L10

But just realise that the repo is outdated...

Other solution could be to update the checks [with](https://www.boost.org/doc/libs/1_72_0/boost/predef/other/endian.h).

Reply:
Wondering if the check would still work with `clang` and Alpine on ARM, [maybe](https://stackoverflow.com/questions/38499462/how-to-tell-clang-to-stop-pretending-to-be-other-compilers)?